### ISSUE TITLE
Resolve case where an app has an `alternate_id` field already defined on its model

### ISSUE THREAD

**Issue by hackmastera** [samvera/hyrax#3618]
When an AF object is converted to a Valkyrie object, if it had alternate_id already defined, we may want to create a new field (`hyrax_alternate_id`?) to use in its stead, copying over the data.

How should we resolve the conflict between hyrax needing to use this field to store the noid and Valkyrie offering it as a reserved field for general use? Should hyrax allow more general use, or reserve it more tightly? Should hyrax advocate for discriminating between different types of alternate_ids?

If we want to reserve it more tightly, we should combine the above behavior with a deprecation warning, asking people to migrate these fields permanently.

**Comment by no-reply** [comment]
notably, Hyrax only needs to use this to store the noid if `Hyrax.config.enable_noids?` is `true`.

when that flag is `false` i think `alternate_id` shouldn't be reserved.

**Comment by tpendragon** [comment]
I can't come up with a proposed success criteria for this one. I don't understand what "downstream use means" - we should talk about this ticket at standup. Valkyrie explicitly made this field to support FCRepo's noid use cases.

**Comment by hackartisan** [comment]
> notably, Hyrax only needs to use this to store the noid if `Hyrax.config.enable_noids?` is `true`.
> 
> when that flag is `false` i think `alternate_id` shouldn't be reserved.

@no-reply good point. If that config defaults to false maybe it's enough to put a comment in the generated configuration file.

The scenario I imagined is that someone has a hyrax app that uses noids, starts storing other values in alternate_id without realizing, and overwrites the noid values and loses them. If that's not a technical concern then let's close this.

**Comment by tpendragon** [comment]
We can depend on downstream applications having test suites that would catch this before they went to production. We think if this was going to be a problem in production applications, it would have happened years ago when we started using Wings.

We recommend closing it.
